src/services/elasticsearch/elasticsearch.service.ts

```diff
@@ -0,0 +1,166 @@
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { throttle } from 'lodash';
+import { Client as ElasticClient } from '@elastic/elasticsearch';
+import { ErrorCause } from '@elastic/elasticsearch/lib/api/types';
+import {
+  DetectedChanges,
+  DetectedChangesType,
+} from '../../util/detect.changes';
+import { ExcalidrawElement } from '../../excalidraw/types';
+import { ELASTICSEARCH_CLIENT_PROVIDER } from '../../elasticsearch-client';
+import {
+  ConfigType,
+  WhiteboardEventLoggingMode,
+  WhiteboardEventLoggingModeType,
+} from '../../config';
+
+type ErroredDocument = {
+  status: number | undefined;
+  error: ErrorCause | undefined;
+  operation: unknown;
+  document: unknown;
+};
+
+type BaseChangeEventDocument = {
+  '@timestamp': Date;
+  createdBy: string;
+  whiteboardId: string;
+  types: DetectedChangesType[];
+};
+
+type LightChangeEventDocument = BaseChangeEventDocument;
```

Comment:
You have `lite` mode but `lightChange...`. That is confusing - it should be either one of the two.